src/Form/GCSettingsForm.php

```diff
@@ -80,7 +81,108 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       '#required' => TRUE,
     ];
 
-    $form['app_settings']['components'] = [
+    $form['app_settings']['switch_legacy_view'] = [
+      '#title' => $this->t('Switch to Legacy View'),
+      '#description' => $this->t('Legacy View has the following components: Virtual Y Video, Live streams, Virtual meetings, Blog posts.'),
+      '#type' => 'checkbox',
+      '#default_value' => $config->get('switch_legacy_view') ?? FALSE,
+    ];
+
+    $this->prepareComponents($form, $config);
+    $this->prepareLegacyView($form, $config);
```

Comment:
```php
    $this->prepareComponents($form, $config);
    $this->prepareLegacyView($form, $config);
```

These functions are very similar. I think we can simplify this and avoid copy/past. Let's add an additional argument `array $components` and maybe container_id and merge to one function

Reply:
We with @Sardis93 decided to rollback to the previous version, so let's resolve this conversation